sentry-ruby/lib/sentry/propagation_context.rb

```diff
@@ -3,15 +3,16 @@
 require "securerandom"
 require "sentry/baggage"
 require "sentry/utils/uuid"
+require "sentry/utils/sample_rand"
 
 module Sentry
   class PropagationContext
     SENTRY_TRACE_REGEXP = Regexp.new(
-      "^[ \t]*" +  # whitespace
+      "\\A[ \\t]{0,10}" +  # limited whitespace at start
```

Comment:
this is unrelated to this PR, please make a separate PR for this and explain the need for changing the regex